Common/lib/CANInterface/CANInterface.cpp

```diff
@@ -1,36 +1,44 @@
 #include "CANInterface.h"
 #include "Printing.h"
+#include "mbed.h"
+#include "rtos.h"
 
 #define CAN_PERIOD  10ms
 
-CANInterface::CANInterface(CAN &c, CANParser &cp, Ticker &tx, DigitalOut *stby, std::chrono::microseconds txPeriod) : can(c), canParser(cp), txTicker(tx)
-{
-    txTicker.attach(callback(this, &CANInterface::txHandler), (txPeriod==0s ? CAN_PERIOD : txPeriod));
-    can.attach(callback(this, &CANInterface::rxHandler), CAN::RxIrq);
-    if(stby)
-    {
+CANInterface::CANInterface(CAN &c, CANParser &cp, Thread &tx_thrd, Thread &rx_thrd, DigitalOut *stby, int txPeriod) : can(c), canParser(cp), tx_thread(rx_thrd), rx_thread(tx_thrd)
+{    
+    if(stby)    // should these be checked in the rx/tx threads? once we initialize this we 
+    {           // won't have control over it anymore 
```

Comment:
In summmary, should we go ahead and just delete this whole if statement, or do you still want to have the 
```
if (stbyI){}
```
Statement in there? 

Also, at this point, if out constructor is empty, would it make sense to actually do object assignment, i.e.
```
CANInterface::CANInterface(CAN &c, CANParser &cp, Thread &tx_thrd, Thread &rx_thrd, DigitalOut *stby, int txPeriod)
{    
    can = c;
    can_parser = cp;

    tx_thread = tx_thrd;
    rx_thread = rx_thrd;
}
```

What's the rationale for using that default constructor operator with 
```
Constructor(args) : class_object(arg1), class_object(arg2), ...
{}
```

Reply:
Sorry that wasn't clear, leave the

```
if (stby) {}
```
in there, so that we are checking against a null pointer (which is the default value)

Reply:
Also, for your second question, we cannot initialize references inside the constructor because references cannot be changed, so we must use the initializer list method. If these were pointers or just regular objects, then initializing them in the constructor would be acceptable, but references do not allow reassignment